### PR TITLE
VS-105: Optimization: When Creating New Syntax Trees, Add All Members

### PR DESCRIPTION
**Pull request by Ravi-Raghavan**
Add all members at once instead of doing it individually.

### CODE REVIEW

File: src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs [hunk 1]
```diff
@@ -69,16 +69,16 @@ public static CompilationResult Compile(MongoAnalysisContext context, Expression
...
         foreach (var builderContext in builderExpressionAnalysis.AnalysisNodeContexts)
         {
-            var analysisNode = builderContext.Node;
-
-            builderContext.EvaluationMethodName = testCodeBuilder.AddBuildersExpression(
-                analysisNode.ArgumentTypeName,
-                analysisNode.RewrittenExpression);
+            var (generatedMqlMethodName, generatedMqlMethodDeclaration) = testCodeBuilder.GenerateMqlGeneratorMethod(builderContext);
+            builderContext.EvaluationMethodName = generatedMqlMethodName;
+            generatedMqlMethodDeclarations.Add(generatedMqlMethodDeclaration);
```
Comment (BorisDog) [inline src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs:78]: It's better to pass down to `GenerateMqlGeneratorMethod` the minimal data it needs, which is `ArgumentTypeName` and `RewrittenExpression`. It should not be aware of `ExpressionAnalysisNode` at this point.

File: src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs [hunk 2]
```diff
@@ -69,16 +69,16 @@ public static CompilationResult Compile(MongoAnalysisContext context, Expression
     private static SyntaxTree GenerateMqlGeneratorSyntaxTree(ExpressionsAnalysis builderExpressionAnalysis)
     {
         var testCodeBuilder = new BuildersMqlGeneratorTemplateBuilder(s_mqlGeneratorSyntaxElements);
+        var generatedMqlMethodDeclarations = new List<MethodDeclarationSyntax>();
```
Comment (BorisDog) [inline src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs:72]: Initialize list size with `builderExpressionAnalysis.AnalysisNodeContexts.Length`

**Review by BorisDog — CHANGES_REQUESTED** [review]
Good idea! Two minor comments.

File: src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs (hunk 1 above)
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs:78]: Done

File: src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs (hunk 2 above)
Comment (Ravi-Raghavan) [reply src/MongoDB.Analyzer/Core/Builders/AnalysisCodeGenerator.cs:72]: Done

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**